packages/@aws-cdk/aws-stepfunctions-tasks/lib/private/task-utils.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  FieldUtils,
+  IntegrationPattern,
+  renderJsonPath,
+  TaskStateBaseProps,
+} from '@aws-cdk/aws-stepfunctions';
+import { Aws } from '@aws-cdk/core';
+
+/**
+ * Represents a service integration call to Step Functions
+ */
+export interface TaskStateConfig extends TaskStateBaseProps {
+  /**
+   * The ARN of resource that represents the work to be executed
+   */
+  readonly resourceArn: string;
+
+  /**
+   * Parameters pass a collection of key-value pairs, either static values or
+   * JSON path expressions that select from the input.
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/dg/input-output-inputpath-params.html#input-output-parameters
+   *
+   * @default - No parameters
+   */
+  readonly parameters?: { [name: string]: any };
+}
+
+/**
+ * Generates the State JSON to define a task state
+ */
+export function taskStateJson(config: TaskStateConfig): any {
```

Comment:
Seems to me that parts of this could go directly into `TaskBase`. Why are they here?

Reply:
it's where I'd started - you asked me to move it 🙃 - i'll put it back

Reply:
your primary concern around putting this into `TaskBase` was around whether this method is something we wanted to make protected, which would mean that it increases the surface area of the base class. 

confirming that it's no longer a problem for you @rix0rrr

Reply:
you're right. I was initially I want to give implementing classes full control to render the task state.
But we can allow that and move the base configuration into `task-base` so going with that.